File: backend/HRManagementSystem/src/Core/HRManagementSystem.Domain/Entities/Goal.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using HRManagementSystem.Domain.Enums.Performance;

namespace HRManagementSystem.Domain.Entities;
public class Goal : BaseEntity
{
    public int PerformanceReviewId { get; set; }
    public string Title { get; set; } = default!;
    public string? Description { get; set; }
    public GoalStatus Status { get; set; } = GoalStatus.NotStarted;
    public decimal ProgressPercent { get; set; } 
    public DateTime? DueDate { get; set; }

    public PerformanceReview Review { get; set; } = default!;
}

File: frontend/src/app/models/performance/goal.ts
import { GoalStatus } from './enums';

export interface Goal {
  id: number;
  performanceReviewId: number;
  title: string;
  description?: string;
  status: GoalStatus;
  progressPercent: number;
  dueDate?: string;
}

export interface CreateGoalDto {
  performanceReviewId: number;
  title: string;
  description?: string;
  dueDate?: string;
}

export interface UpdateGoalProgressDto {
  goalId: number;
  progressPercent: number;
  status?: string;
}
